core-backend/wasmi/src/funcs/mod.rs

```diff
@@ -430,15 +430,13 @@ where
         at: u32,
         len: u32,
     ) -> Result<&'_ [u8], FuncError<<E as Ext>::Error>> {
-        let msg = ext.read().map_err(FuncError::Core)?;
+        let msg = ext.read(at, len).map_err(FuncError::Core)?;
 
-        let last_idx = at
-            .checked_add(len)
-            .ok_or_else(|| FuncError::ReadLenOverflow(at, len))?;
+        // 'at' and 'len' correct and saturation checked in Ext::read
+        assert!(at.checked_add(len).is_some());
+        let last_idx = at + len;
 
-        if last_idx as usize > msg.len() {
-            return Err(FuncError::ReadWrongRange(at..last_idx, msg.len() as u32));
-        }
+        assert!(last_idx as usize > msg.len());
```

Comment:
asserts work in Rust even for release builds, so it seems better to eliminate them completely or use `debug_assert`